support-models/src/main/scala/com/gu/support/workers/redemption/RedemptionData.scala

```diff
@@ -0,0 +1,35 @@
+package com.gu.support.workers.redemption
+
+import cats.syntax.functor._
+import com.gu.support.encoding.Codec
+import com.gu.support.zuora.api.{ReaderType, Subscription}
+import io.circe.syntax._
+import io.circe.{Decoder, Encoder}
+
+abstract class RedemptionData(val redemptionCode: RedemptionCode) {
+  def redeem(subscription: Subscription) = subscription.copy(redemptionCode = Some(redemptionCode))
+}
+
+case class Corporate(override val redemptionCode: RedemptionCode, corporateAccountId: CorporateAccountId) extends RedemptionData(redemptionCode) {
```

Comment:
we also have `com.gu.support.catalog.Corporate`, consider being a bit more verbose in the class name rather than just reling on package name?